client/src/services/resourceService.test.js

```diff
@@ -41,6 +43,49 @@ describe("ResourceService", () => {
 		});
 	});
 
+	describe("getDrafById", () => {
+		const resourceID = randomUUID();
+
+		it("should return resource correctly", async () => {
+			const mockResource = resourceStub({
+				recommender: randomUUID(),
+				id: resourceID,
+				accession: "2023-09-02T08:19:30.771Z",
+			});
+
+			server.use(
+				rest.get(`/api/resources/${resourceID}`, (_, res, ctx) => {
+					return res(ctx.json(mockResource));
+				})
+			);
+
+			const expectedResource = await service.getDraftById(resourceID);
+
+			expect(expectedResource.data).toEqual(mockResource);
+			expect(expectedResource.error).toBeUndefined();
+		});
+
+		it("should handle server error", async () => {
+			server.use(
+				rest.get(`/api/resources/${resourceID}`, (_, res, ctx) => {
+					return res(ctx.status(503));
+				})
+			);
+
+			const expectedResource = await service.getDraftById(resourceID);
+
+			expect(expectedResource.data).toBeUndefined();
+			expect(expectedResource.error).toBe("error with status 503");
+		});
+
+		it("should handle exception error", async () => {
+			const response = await service.getDraftById("unknownID");
+
+			expect(response.data).toBeUndefined();
+			expect(response.error).toBeDefined();
+		});
```

Comment:
its for handling network request of fetch, the "unknownID" is misleading, you are right :D

Reply:
I think you're using the implicit _lack_ of handling from MSW, with `onUnhandledRequest` throwing an error, to get the failure state here. In cases like these I'd recommend using an explicit https://mswjs.io/docs/api/response/network-error instead, so that the details of what's intended are within the test body (and the test doesn't start failing incorrectly if someone adds a default handler later).

Reply:
good tip thanks